process/marshal.go

```diff
@@ -486,6 +558,10 @@ func (inbox *Inbox) UnmarshalJSON(data []byte) error {
 					msg := new(Precommit)
 					err = json.Unmarshal(raw, msg)
 					inbox.messages[height][round][sig] = msg
+				case ResyncMessageType:
```

Comment:
There is probably no need to store resync messages inside the inbox. The only situation in which this would be necessary is when the receiver crashes before responding to the resync. However, given that the sender only needs one receiver to respond, this kind of fault is acceptable (the sender can send the resync message to a few nodes, and the likelihood that all of them will crash before responding is low).

Reply:
We don't actually store `Resync` messages. If you have a look at the `State` type, we expect an inbox for `Propose`, `Prevote` and `Precommit` messages. The marshaler just implements (un)marshaling for all types, without any assumptions about what to expect.